pkg/cluster/cluster_info_scraper.go

```diff
@@ -0,0 +1,169 @@
+package cluster
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	client "k8s.io/client-go/kubernetes"
+	api "k8s.io/client-go/pkg/api/v1"
+	restclient "k8s.io/client-go/rest"
+
+	"github.com/golang/glog"
+)
+
+const (
+	k8sDefaultNamespace = "default"
+
+	kubernetesServiceName = "kubernetes"
+)
+
+var (
+	labelSelectEverything = labels.Everything().String()
+	fieldSelectEverything = fields.Everything().String()
+)
+
+type ClusterScraper struct {
+	*client.Clientset
+}
+
+func NewClusterInfoScraper(kubeConfig *restclient.Config) (*ClusterScraper, error) {
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kubeClient: %s", err)
+	}
+
+	return &ClusterScraper{
+		Clientset: kubeClient,
+	}, nil
+}
+
+func (s *ClusterScraper) GetAllNodes() ([]*api.Node, error) {
+	listOption := metav1.ListOptions{
+		LabelSelector: labelSelectEverything,
+		FieldSelector: fieldSelectEverything,
+	}
+	return s.GetNodes(listOption)
+}
+
+func (s *ClusterScraper) GetNodes(opts metav1.ListOptions) ([]*api.Node, error) {
+	nodeList, err := s.CoreV1().Nodes().List(opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list all nodes in the cluster: %s", err)
+	}
+	var nodes []*api.Node
+	for _, n := range nodeList.Items {
+		node := n
+		nodes = append(nodes, &node)
+	}
+
+	return nodes, nil
+}
+
+func (s *ClusterScraper) GetAllPods() ([]*api.Pod, error) {
+	listOption := metav1.ListOptions{
+		LabelSelector: labelSelectEverything,
+		FieldSelector: fieldSelectEverything,
+	}
+	return s.GetPods(api.NamespaceAll, listOption)
+}
+
+func (s *ClusterScraper) GetPods(namespaces string, opts metav1.ListOptions) ([]*api.Pod, error) {
+	podList, err := s.CoreV1().Pods(namespaces).List(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	pods := []*api.Pod{}
```

Comment:
append is kind of less efficient. 
Since the total length of the array is known here,  it is better call make([]*api.Pod, length).